src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -1625,39 +1621,14 @@ public sealed override async Task<HttpResponseMessage> SendAsync(HttpRequestMess
             }
             catch (Exception e)
             {
-                Exception replacementException = null;
-
                 if (e is IOException ||
                     e is ObjectDisposedException ||
                     e is Http2ProtocolException ||
                     e is InvalidOperationException)
                 {
-                    replacementException = new HttpRequestException(SR.net_http_client_execution_error, _abortException ?? e);
-                }
-                else if (e is OperationCanceledException oce)
-                {
-                    // If the operation has been canceled after the stream was allocated an ID, send a RST_STREAM.
-                    if (http2Stream != null && http2Stream.StreamId != 0)
-                    {
-                        http2Stream.Cancel();
-                    }
```

Comment:
It's ok to remove this because the RST_STREAM will always be triggered elsewhere?

Reply:
Yes. RST_STREAM will be sent when (a) both request and response complete (so we're guaranteed to not try to send any more frames on this stream; (b) one or both have failed; (c) we haven't received a reset from the server. See CheckForCompletion.